pkgs/development/libraries/libnotify/default.nix

```diff
@@ -27,15 +28,24 @@ stdenv.mkDerivation rec {
     "-Dtests=false"
     "-Ddocbook_docs=disabled"
     "-Dgtk_doc=false"
+    "-Dintrospection=${if withIntrospection then "enabled" else "disabled"}"
   ];
 
+  strictDeps = true;
+
   nativeBuildInputs = [
-    gobject-introspection
     meson
     ninja
     pkg-config
     libxslt
     docbook-xsl-ns
+    glib # for glib-mkenums needed during the build
+  ] ++ lib.optionals withIntrospection [
+    gobject-introspection
+  ];
+
+  buildInputs = lib.optionals withIntrospection [
```

Comment:
Why in `buildInputs`?

Reply:
if i apply
 
```diff
diff --git a/pkgs/development/libraries/libnotify/default.nix b/pkgs/development/libraries/libnotify/default.nix
index 40089e3d7e9..cf5dd487e75 100644
--- a/pkgs/development/libraries/libnotify/default.nix
+++ b/pkgs/development/libraries/libnotify/default.nix
@@ -33,6 +33,10 @@ stdenv.mkDerivation rec {
 
   strictDeps = true;
 
+  depsBuildBuild = [
+    pkg-config
+  ];
+
   nativeBuildInputs = [
     meson
     ninja
@@ -44,10 +48,6 @@ stdenv.mkDerivation rec {
     gobject-introspection
   ];
 
-  buildInputs = lib.optionals withIntrospection [
-    gobject-introspection
-  ];
-
   propagatedBuildInputs = [
     gdk-pixbuf
     glib
```

then the build fails with 
```
libnotify> [14/15] Generating Notify-0.7.gir with a custom command0.p/libnotify.so.4.0.0.symbols
libnotify> FAILED: libnotify/Notify-0.7.gir
libnotify> /nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/bin/g-ir-scanner --no-libtool --namespace=Notify --nsversion=0.7 --warn-all --output libnotify/Notify-0.7.gir --c-include=libnotify/notify.h -I/build/libnotify-0.7.9/libnotify -I/build/libnotify-0.7.9/build/libnotify -I./. -I../. --filelist=/build/libnotify-0.7.9/build/libnotify/libnotify.so.4.0.0.p/Notify_0.7_gir_filelist --include=GLib-2.0 --include=GdkPixbuf-2.0 --pkg-export=libnotify --cflags-begin -I./. -I../. '-DG_LOG_DOMAIN="libnotify"' -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include/glib-2.0 -I/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib/glib-2.0/include -I/nix/store/1ji5kw304sdwvdb9asqniwmrdmkx3rxg-gdk-pixbuf-2.42.6-dev/include/gdk-pixbuf-2.0 -I/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/include/gobject-introspection-1.0 --cflags-end --add-include-path=/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0 -L./libnotify --library notify -L/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib -L/nix/store/xk2is79zbvmslhb1bw4dk12j1jbyy2l6-gdk-pixbuf-2.42.6/lib --extra-library=gdk_pixbuf-2.0 --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=gio-2.0 -L/nix/store/rd8na9jgqgvp4k1razyxlxdi15416l57-gobject-introspection-1.70.0/lib --extra-library=girepository-1.0 --sources-top-dirs /build/libnotify-0.7.9/subprojects/ --sources-top-dirs /build/libnotify-0.7.9/build/subprojects/
libnotify> Couldn't find include 'GdkPixbuf-2.0.gir' (search path: '['/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share', '/nix/store/dvjlviry8rrcrgaxj63q95wxzxagjbpg-pkg-config-wrapper-0.29.2/share/gir-1.0', '/nix/store/fvwk5kbk1rifxjva4j2j40isj80v5yb7-meson-0.57.1/share/gir-1.0', '/nix/store/y5v8rk457phrv71jc9slx3a58lyvdmll-ninja-1.10.2/share/gir-1.0', '/nix/store/1dr9x27m5mxqrv0rx2i9ckz9m34gkvfw-libxslt-1.1.34-dev/share/gir-1.0', '/nix/store/bzaw10zf10wf3ws5xp59zkvmp0xg0chn-libxslt-1.1.34-bin/share/gir-1.0', '/nix/store/kybxplf3r7rbbaajsr4zwwvcbj23dxf8-docbook-xsl-ns-1.79.2/share/gir-1.0', '/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/share/gir-1.0', '/nix/store/gxwpxk04i6appwhv5zsmx2rm4zjr10bd-zlib-1.2.11/share/gir-1.0', '/nix/store/vw7v3r8zgxp5kgc6m736zb98ybbrcfa4-gettext-0.21/share/gir-1.0', '/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/c7mwhb56d18aj4nl0g2hs2203cn6z5pk-patchelf-0.13/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/usr/share/gir-1.0']')
libnotify> ninja: build stopped: subcommand failed.
```

Reply:
Hmm, that sounds like the `gobject-introspection` setup hook not setting `XDG_DATA_DIRS` properly.